src/mapping.ts

```diff
@@ -1,410 +1,477 @@
+   * Checks if to/from addresses are not one of Valorem's or the Zero Address
+   * If neither are, records the amounts transferred in daily metrics
+   */
+  // get params
+  const fromAddress = event.params.from.toHexString();
+  const toAddress = event.params.to.toHexString();
+  const tokenId = event.params.id;
+  const amount = event.params.amount;
+
+  // check transfer isn't an OSE Write/Exercise/Redeem event (mint/OSE is receiver/burn)
+  const transferIsExternal =
+    fromAddress != ZERO_ADDRESS &&
+    toAddress != ZERO_ADDRESS &&
+    toAddress != event.address.toHexString();
+
+  if (transferIsExternal) {
+    // bind to OptionSettlementEngine to call view functions
+    const ose = OSEContract.bind(Address.fromBytes(event.address));
+    // get type of Token
+    const tokenType = ose.tokenType(tokenId);
+
+    handleERC1155TransferMetrics(tokenId, amount, event, tokenType);
+  }
 }
 
-export function handleOptionsWritten(event: OptionsWritten): void {
-  let contract = fetchERC1155(event.address);
-  let token = fetchERC1155Token(contract, event.params.claimId);
-  token.claim = event.params.claimId.toString();
-  token.type = 2;
-  token.save();
-
-  let option = Option.load(event.params.optionId.toString())!;
-  // should never be null. if option doesnt exist, somethings wrong with handleNewOptionType
-  // if (option == null) {
-  //   option = new Option(event.params.optionId.toString());
-  //   option.save();
-  // }
-
-  const underlyingAssetAddress = option.underlyingAsset
-    ? option.underlyingAsset
-    : ZERO_ADDRESS;
-
-  const underlyingPriceUSD = getTokenPriceUSD(underlyingAssetAddress as string);
+// https://github.com/OpenZeppelin/openzeppelin-subgraphs
+export function handleTransferBatch(event: TransferBatchEvent): void {
+  const contract = fetchERC1155(event.address.toHexString());
+  const operator = fetchAccount(event.params.operator.toHexString());
+  const from = fetchAccount(event.params.from.toHexString());
+  const to = fetchAccount(event.params.to.toHexString());
+  const ids = event.params.ids;
+  const amounts = event.params.amounts;
+  // If this equality doesn't hold (some devs actually don't follow the ERC specifications) then we just can't make
+  // sense of what is happening. Don't try to make something out of stupid code, and just throw the event. This
+  // contract doesn't follow the standard anyway.
+  if (ids.length == amounts.length) {
+    for (let i = 0; i < ids.length; ++i) {
+      registerTransfer(
+        event,
+        "/".concat(i.toString()),
+        contract,
+        operator,
+        from,
+        to,
+        ids[i],
+        amounts[i]
+      );
+    }
+  }
 
-  const underlyingAmount = (option.underlyingAmount as BigInt)
-    .toBigDecimal()
-    .div(
-      exponentToBigDecimal(
-        BigInt.fromI64(
-          ERC20.bind(
-            Address.fromString(underlyingAssetAddress as string)
-          ).decimals()
-        )
-      )
-    );
+  /**
```

Comment:
Anyway we can de-dupe this Valorem Extension code to DRY things up?

Reply:
I tried to keep the notice close to the changes, so in the future anyone looking will be able to tell what's OZ/what's Valorem's. What do you have in mind?

Reply:
- DRY up the code itself.
- Rename to `isMintOrBurn(from, to)`.

Reply:
refactored here: https://github.com/valorem-labs-inc/valorem-subgraph/commit/59eefb0dc693d6b4f3a5d88582dcd7335a26a7a4